lib/matrix.js

```diff
@@ -52,11 +58,351 @@ var init = function(exports){
+            
+            var displayName;
+            var memberEvent = this.store.getStateEvent(roomId, 'm.room.member', userId);
+            if (memberEvent && memberEvent.event.content.displayname) {
+                displayName = memberEvent.event.content.displayname;
+            }
+            else {
+                return userId;
+            }
+            
+            var members = this.store.getStateEvents(roomId, 'm.room.member')
+                .filter(function(event) {
+                    return event.event.content.displayname === displayName;
+                });
+                
+            if (members.length > 1) {
+                return displayName + " (" + userId + ")";
+            }
+            else {
+                return displayName;
+            }
+        },
+
+        /*
+         * High level helper method to call initialSync, emit the resulting events,
+         * and then start polling the eventStream for new events.
+         */
+        startClient: function(callback, historyLen) {
+            historyLen = historyLen || 12;
+            
+            var self = this;
+            if (!this.fromToken) {
+                this.initialSync(historyLen, function(err, data) {
+                    if (err) {
+                        console.error("err %s", JSON.stringify(err));
+                        callback(err);
+                    } else {
+                        var events = []
+                        for (var i = 0; i < data.presence.length; i++) {
+                            events.push(new MatrixEvent(data.presence[i]));
+                        }
+                        for (var i = 0; i < data.rooms.length; i++) {
+                            for (var j = 0; j < data.rooms[i].state.length; j++) {
+                                events.push(new MatrixEvent(data.rooms[i].state[j]));
+                            }
+                            for (var j = 0; j < data.rooms[i].messages.chunk.length; j++) {
+                                events.push(new MatrixEvent(data.rooms[i].messages.chunk[j]));
+                            }
+                        }
+                        callback(undefined, events, false);
+                        self.clientRunning = true;
+                        self._pollForEvents(callback);
+                    }
+                });
+            }
+            else {
+                this._pollForEvents(callback);
+            }
+        },
+        
+        _pollForEvents: function(callback) {
+            var self = this;
+            if (!this.clientRunning) return;
+            this.eventStream(this.fromToken, 30000, function(err, data) {
+                if (err) {
+                    console.error("err %s", JSON.stringify(err));
+                    callback(err);
+                    // retry every few seconds
+                    setTimeout(function() {
+                        self._pollForEvents(callback);
+                    }, 2000);
```

Comment:
Given this SDK is pretty fundamental to a lot of clients, we should really be doing exp backoff ASAP. I would propose a high cap of something like 1 *hour* (bear in mind that IM isn't the only use case here) but crucially provide a manual nudge function which will kill the timer and do another poll so humans aren't at the mercy of backoff. This can wait for another PR given it's separate and a bit fiddly.